src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -80,18 +84,132 @@ protected override bool ValidateParameters()
             return true;
         }
 
-        protected override string GenerateCommandLineCommands()
+        #region TEMP LOGIC - SDK Should provide correct list - Delete this when fixing https://github.com/dotnet/sdk/issues/3110
+        bool IsManagedAssemblyToUseAsCrossgenReference(ITaskItem file)
+        {
+            // Reference only managed assemblies that will be published to the root directory.
+            string relativeOutputPath = file.GetMetadata(MetadataKeys.RelativePath);
+            if (!String.IsNullOrEmpty(Path.GetDirectoryName(relativeOutputPath)))
+            {
+                return false;
+            }
+
+            using (FileStream fs = new FileStream(file.ItemSpec, FileMode.Open, FileAccess.Read))
+            {
+                try
+                {
+                    using (var pereader = new PEReader(fs))
+                    {
+                        if (!pereader.HasMetadata)
+                            return false;
+
+                        MetadataReader mdReader = pereader.GetMetadataReader();
+                        if (!mdReader.IsAssembly)
+                            return false;
+
+                        // Reference assemblies should never be given to crossgen
+                        if (IsReferenceAssembly(mdReader))
+                            return false;
+                    }
+                }
+                catch (BadImageFormatException)
+                {
+                    // Not a valid assembly file
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool IsReferenceAssembly(MetadataReader mdReader)
```

Comment:
I know you're saying this is temporary, but why do we need to duplicate what the prepare step checks here? Why can't we handle building the right list in the prepare step?